core/include/macros/units.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_macros
+ * @{
+ *
+ * @file
+ * @brief       Unit helper macros
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+
+#ifndef UNITS_H
+#define UNITS_H
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/* KiB, MiB, GiB */
+#define KiB(x) ((unsigned long)(x) << 10)
+#define MiB(x) (KiB(x) << 10)
+#define GiB(x) (MiB(x) << 10)
```

Comment:
Just to confirm - this only works for up to 3g, right? (because `4 = 0b100`, and `0b100 << 30 = 0b1[32 zeros]`, which overflows unsigned int, which KiB has been force-casted to.)

If we define `KiB()` as `unsigned long long`, will the optimizer do the right thing?

Reply:
Or just `#define GiB(x) ((unsigned long long)MiB(x) << 10)`?

Reply:
Would 4194 be enough for MiB?